lib/src/preview/client.dart

```diff
@@ -0,0 +1,62 @@
+import 'dart:typed_data';
+
+import 'package:http/http.dart';
+
+import '../network.dart';
+
+/// PreviewClient abstracts preview and thumbnail API
+class PreviewClient {
+  // ignore: public_member_api_docs
+  PreviewClient(
+    this._baseUrl,
+    String username,
+    String password
+  ) {
+    final client = NextCloudHttpClient(username, password);
+    _network = Network(client);
+  }
+
+  final String _baseUrl;
+  Network _network;
+
+  String _getPreviewUrl(String remotePath, int width, int height) {
+    return '$_baseUrl/index.php/core/preview.png?file=${Uri.encodeQueryComponent(remotePath)}&x=$width&y=$height&a=1&mode=cover&forceIcon=0';
+  }
+
+  String _getThumbnailUrl(String remotePath, int width, int height) {
+    if(!remotePath.startsWith('/')) {
+      remotePath = '/$remotePath';
+    }
+    return '$_baseUrl/index.php/apps/files/api/v1/thumbnail/$width/$height${Uri.encodeFull(remotePath)}';
+  }
+
+  /// fetch preview for [remotePath] with provided [width] and [height]
+  /// [width] and [height] default to 512
+  Future<Uint8List> getPreview(String remotePath, {int width = 512, int height = 512}) async {
```

Comment:
Are these default values also given in the java library? If not I would remove them so the user has to explicitly specify the dimensions.

Reply:
I removed the default values. Sizes are now mandatory.